### PR TITLE
Align gcloud installation (Resubmit #1155)

### PR DESCRIPTION
**Pull request by Michaelvll**
This is a minor change to align the installation code for our cloud store and spot controller.

The previous PR has a problem with the generated spot controller YAML due to the incorrect indent. It is now fixed by not using multiple lines for the installation commands.

Tested:
- [x] `sky spot launch -n test 'sleep 10'`
- [x] `sky launch --cloud gcp examples/using_file_mounts.yaml`
- [x] `sky launch examples/using_file_mounts.yaml`

### CODE REVIEW

File: sky/cloud_stores.py [hunk 1]
```diff
@@ -93,26 +94,16 @@ class GcsCloudStorage(CloudStorage):
...
 
-    _GSUTIL = '~/google-cloud-sdk/bin/gsutil'
+    _GSUTIL = 'gsutil'
 
     def is_directory(self, url: str) -> bool:
         """Returns whether 'url' is a directory.
         In cloud object stores, a "directory" refers to a regular object whose
         name is a prefix of other objects.
         """
-        commands = list(self._GET_GSUTIL)
```
Comment (Michaelvll) [inline sky/cloud_stores.py:115]: The previous code will cause a problem that the commands appended will be accumulated in the `self._GET_GSUTIL`...

**Review by romilbhardwaj — APPROVED** [review]
lgtm! thanks for quickly identifying the bug!

**Comment by Michaelvll** [comment]
Tested again with the 3 tests in the description. I modified the backup logic, so that the `~/.sky/.sky_gcp_config_default_backup` will not be empty for the spot cluster when launched with `sky spot launch`. PTAL. : )

**Comment by Michaelvll** [comment]
Tested:
- [ ] `tests/run_smoke_tests.sh`

**Pull request merged**